### ISSUE TITLE
ContentProvider notifications is not working

### ISSUE THREAD

**Issue by vovkab** [requery/sqlite-android#9]
Subj,

is it because all notifications is stripped out from the code or there is something else?

AbstractCursor:
https://github.com/requery/sqlite-android/blob/master/sqlite-android/src/main/java/io/requery/android/database/AbstractCursor.java

```
    @Override
    public void setNotificationUri(ContentResolver cr, Uri notifyUri) {
    }

    @Override
    public Uri getNotificationUri() {
        return null;
    }
```

while in original code they have this:
https://github.com/android/platform_frameworks_base/blob/4535e11fb7010f2b104d3f8b3954407b9f330e0f/core/java/android/database/AbstractCursor.java

```
    @Override
    public void setNotificationUri(ContentResolver cr, Uri notifyUri) {
        setNotificationUri(cr, notifyUri, UserHandle.myUserId());
    }

    /** @hide - set the notification uri but with an observer for a particular user's view */
    public void setNotificationUri(ContentResolver cr, Uri notifyUri, int userHandle) {
        synchronized (mSelfObserverLock) {
            mNotifyUri = notifyUri;
            mContentResolver = cr;
            if (mSelfObserver != null) {
                mContentResolver.unregisterContentObserver(mSelfObserver);
            }
            mSelfObserver = new SelfContentObserver(this);
            mContentResolver.registerContentObserver(mNotifyUri, true, mSelfObserver, userHandle);
            mSelfObserverRegistered = true;
        }
    }

    @Override
    public Uri getNotificationUri() {
        synchronized (mSelfObserverLock) {
            return mNotifyUri;
        }
    }
```

**Comment by npurushe** [comment]
Yeah this was because the mContentResolver was marked deprecated an there was a dependency on the non-public UserHandle. Since setNotificationUri isn't deprecated and it looks like UserHandle api isn't needed will fix.

**Comment by vovkab** [comment]
@npurushe thank you for a quick fix!
When do you guys planning to do next release? :)

**Comment by npurushe** [comment]
Could be a few days. However this change is made in the snapshot version, you can use this change right away by using:

```gradle
    repositories {
        jcenter()
        maven { url 'http://oss.jfrog.org/artifactory/oss-snapshot-local' }
    }
```

```gradle
dependencies {
    compile 'io.requery:sqlite-android:3.12.1-1-SNAPSHOT'
}
```

**Comment by vovkab** [comment]
@npurushe nice, thanks for a quick response!

**Comment by vovkab** [comment]
Tried updated version, notifications still not working, seems like something else is missing :(

**Comment by npurushe** [comment]
Hmm the changed code just register itselfs as `ContentObserver` on the `ContentResolver`, if you put a breakpoint there and step into it do you see it get registered?

**Comment by vovkab** [comment]
@npurushe created pull request for a fix #10

**Comment by npurushe** [comment]
Good find Thanks!
